metadata-ingestion/src/datahub/ingestion/transformer/add_dataset_browse_path.py

```diff
@@ -0,0 +1,62 @@
+from typing import List
+
+import datahub.emitter.mce_builder as builder
+from datahub.configuration.common import ConfigModel
+from datahub.ingestion.api.common import PipelineContext
+from datahub.ingestion.transformer.dataset_transformer import DatasetTransformer
+from datahub.metadata.schema_classes import (
+    BrowsePathsClass,
+    DatasetSnapshotClass,
+    MetadataChangeEventClass,
+)
+
+
+class AddDatasetBrowsePathConfig(ConfigModel):
+    path_templates: List[str]
+
+
+class AddDatasetBrowsePathTransformer(DatasetTransformer):
+    """Transformer that can be used to set browse paths through template replacement"""
+
+    ctx: PipelineContext
+    config: AddDatasetBrowsePathConfig
+
+    def __init__(self, config: AddDatasetBrowsePathConfig, ctx: PipelineContext):
+        self.ctx = ctx
+        self.config = config
+
+    @classmethod
+    def create(
+        cls, config_dict: dict, ctx: PipelineContext
+    ) -> "AddDatasetBrowsePathTransformer":
+        config = AddDatasetBrowsePathConfig.parse_obj(config_dict)
+        return cls(config, ctx)
+
+    def transform_one(self, mce: MetadataChangeEventClass) -> MetadataChangeEventClass:
+        if not isinstance(mce.proposedSnapshot, DatasetSnapshotClass):
```

Comment:
this handling can be moved up to the base class `DatasetTransformer`

Reply:
Will do

Reply:
Moving it caused `mypy` errors https://github.com/linkedin/datahub/runs/3276902373#step:5:838. So reverting this change to move up to base class

Reply:
Ok, there might be a way to do this, but we can take care of it later.